war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -2923,18 +2923,24 @@ function applyErrorMessage(elt, rsp) {
     if (rsp.status == 200) {
         elt.innerHTML = rsp.responseText;
     } else {
-        var id = 'valerr' + (iota++);
-        elt.innerHTML = '<a href="" onclick="document.getElementById(\'' + id
-        + '\').style.display=\'block\';return false">ERROR</a><div id="'
-        + id + '" style="display:none">' + rsp.responseText + '</div>';
-        var error = document.getElementById('error-description'); // cf. oops.jelly
-        if (error) {
-            var div = document.getElementById(id);
-            while (div.firstElementChild) {
-                div.removeChild(div.firstElementChild);
-            }
-            div.appendChild(error);
-        }
+        var parser = new DOMParser();
+        var doc = parser.parseFromString(rsp.responseText, 'text/html');
```

Comment:
We should also check that the response is in fact HTML and not some other Content-Type. The original code also didn’t do this so I’m unsure if there are any endpoints that don’t respond with HTML in some cases.